### PR TITLE
SONARJAVA-1762 division by zero check

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-frontend/src/test/files/se/DivisionByZeroCheck.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    u %= p; // Compliant
+    return 0;
+  }
+
+  int car(int s) {
+    if (s >= 0) {}
+
+    if (s <= 0) {
+    } else {
+      int x = 14 / s; // Noncompliant - False Positive !!
```
Comment (m-g-sonar) [inline java-frontend/src/test/files/se/DivisionByZeroCheck.java]: I'm still working on this issue, which corresponds to cases remaining in ruling JDK6.

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+import java.util.List;
+
+@Rule(key = "S3518")
+public class DivisionByZeroCheck extends SECheck {
+
+  private enum Status {
+    ZERO, NON_ZERO;
+  }
+
+  private static class DivisionByZeroSymbolicValue extends SymbolicValue {
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: MAJOR Override this superclass' "equals" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2160)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [BinarySymbolicValue.java#L54](https://github.com/SonarSource/sonar-java/blob/45fe923491b9ac275f5608ca3ab947281ff23cfb/java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/BinarySymbolicValue.java#L54): The Cyclomatic Complexity of this method "copyConstraint" is 11 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: java-frontend/src/test/files/se/DivisionByZeroCheck.java [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+    return 0;
+  }
+
+  int car(int s) {
+    if (s >= 0) {
+    }
+
+    if (s <= 0) {
+    } else {
+      int x = 14 / s; // Noncompliant - FP
```
Comment (m-g-sonar) [inline java-frontend/src/test/files/se/DivisionByZeroCheck.java]: still working on these ones...

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [BinarySymbolicValue.java#L54](https://github.com/SonarSource/sonar-java/blob/17e86fe601d77eb93fe86c2dbf6f7268fb73891e/java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/BinarySymbolicValue.java#L54): The Cyclomatic Complexity of this method "copyConstraint" is 11 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+          addZeroConstraint(varIsZero, tree);
+          break;
+        default:
+          // do nothing
+          break;
+      }
+    }
+
+    @Override
+    public void visitBinaryExpression(BinaryExpressionTree tree) {
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: MAJOR The Cyclomatic Complexity of this method "visitBinaryExpression" is 13 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [BinarySymbolicValue.java#L54](https://github.com/SonarSource/sonar-java/blob/4e705b915815ea7a1cd79b8b4eb91901f3fddbf7/java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/BinarySymbolicValue.java#L54): The Cyclomatic Complexity of this method "copyConstraint" is 11 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+    @Override
+    public String toString() {
+      // For debugging purpose
+      return "" + (isZero() ? Status.ZERO : Status.NON_ZERO);
+    }
+  }
+
+  @Override
+  public ProgramState checkPreStatement(CheckerContext context, Tree syntaxNode) {
+    if (syntaxNode.is(Tree.Kind.DIVIDE_ASSIGNMENT, Tree.Kind.DIVIDE, Tree.Kind.REMAINDER_ASSIGNMENT, Tree.Kind.REMAINDER)) {
```
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: if you have a prestatement visitor, why having this check  before ?

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+  }
+
+  private class PreStatementVisitor extends CheckerTreeNodeVisitor {
+
+    PreStatementVisitor(CheckerContext context) {
+      super(context.getState());
+    }
+
+    private void addZeroConstraint(boolean isZero, Tree tree) {
+      initialConstraints.put(tree, isZero ? ZeroConstraint.zero(tree) : ZeroConstraint.nonZero(tree));
```
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: As discussed, this is utterly wrong : something that has not the zero constraint is not necessarily nonZero.

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+      if (tree.is(Tree.Kind.INT_LITERAL, Tree.Kind.LONG_LITERAL, Tree.Kind.DOUBLE_LITERAL, Tree.Kind.FLOAT_LITERAL)) {
+        SymbolicValue sv = programState.peekValue();
+        ZeroConstraint constraint = isNumberZero(tree.value()) ? ZeroConstraint.zero(tree) : ZeroConstraint.nonZero(tree);
+        programState = programState.addConstraint(sv, constraint);
+      }
+    }
+
+
+    private boolean isNumberZero(String literalValue) {
+      return !literalValue.matches(".*[1-9]+.*") || (literalValue.startsWith("0x") && !literalValue.substring(2).matches(".*[1-9a-fA-F]+.*"));
```
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: I am wondering why this one is that complicated ?

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+import java.util.Map;
+
+@Rule(key = "S3518")
+public class DivisionByZeroCheck extends SECheck {
+
+  private enum Status {
+    ZERO, NON_ZERO;
+  }
+
+  private Map<Tree, ZeroConstraint> initialConstraints = new HashMap<>();
```
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: This should not exists. management of tree and constraints should be out of scope of the check.

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java (hunk 7 above)
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: a lot of cases won't be covered : byte and char for instance.

**Comment by benzonico** [comment]
As this check is contributing to the flow of SE engine, even if the rule is not activated, it should be run.

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+  private enum Status {
+    ZERO, NON_ZERO;
+  }
+
+  @Override
+  public void scanFile(JavaFileScannerContext context) {
+    super.scanFile(context);
+  }
+
+  private static class ZeroSymbolicValue extends SymbolicValue {
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: MAJOR Override this superclass' "equals" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2160)

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java [hunk 10]
```diff
@@ -0,0 +1,261 @@
...
+  }
+
+  @Override
+  public ProgramState checkPostStatement(CheckerContext context, Tree syntaxNode) {
+    PostStatementVisitor visitor = new PostStatementVisitor(context);
+    syntaxNode.accept(visitor);
+    return visitor.programState;
+  }
+
+  private class PostStatementVisitor extends CheckerTreeNodeVisitor {
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: MAJOR Make this a "static" inner class. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2694)

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java [hunk 11]
```diff
@@ -0,0 +1,261 @@
...
+
+@Rule(key = "S3518")
+public class DivisionByZeroCheck extends SECheck {
+
+  private enum Status {
+    ZERO, NON_ZERO;
+  }
+
+  @Override
+  public void scanFile(JavaFileScannerContext context) {
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: MINOR Remove this method to simply inherit it. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1185)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues
* MAJOR 4 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [BinarySymbolicValue.java#L54](https://github.com/SonarSource/sonar-java/blob/3e496e255183b524639f7a10265c06845ec97a1c/java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/BinarySymbolicValue.java#L54): The Cyclomatic Complexity of this method "copyConstraint" is 11 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java [hunk 12]
```diff
@@ -0,0 +1,258 @@
...
+      switch (tree.kind()) {
+        case MULTIPLY_ASSIGNMENT:
+          addZeroConstraint(exprIsZero);
+          break;
+        case MINUS_ASSIGNMENT:
+        case PLUS_ASSIGNMENT:
+          addZeroConstraint(varIsZero);
+          break;
+        case DIVIDE_ASSIGNMENT:
+        case REMAINDER_ASSIGNMENT:
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: MAJOR Reduce this switch case number of lines from 6 to at most 5, for example by extracting code into methods. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1151)

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java [hunk 13]
```diff
@@ -0,0 +1,258 @@
...
+        switch (tree.kind()) {
+          case MULTIPLY:
+            addZeroConstraint(rightOpIsZero || leftOpIsZero);
+            break;
+          case PLUS:
+          case MINUS:
+            addZeroConstraint(rightOpIsZero && leftOpIsZero);
+            break;
+          case DIVIDE:
+          case REMAINDER:
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: MAJOR Reduce this switch case number of lines from 6 to at most 5, for example by extracting code into methods. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1151)

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 issues
* MAJOR 5 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [BinarySymbolicValue.java#L54](https://github.com/SonarSource/sonar-java/blob/b7068dd2a340feb3618988bd99c0330426638b2e/java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/BinarySymbolicValue.java#L54): The Cyclomatic Complexity of this method "copyConstraint" is 11 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java [hunk 14]
```diff
@@ -0,0 +1,299 @@
...
+        case PLUS:
+        case PLUS_ASSIGNMENT:
+        case MINUS:
+        case MINUS_ASSIGNMENT:
+          handlePlusMinus(leftOp, rightOp);
+          break;
+        case DIVIDE:
+        case DIVIDE_ASSIGNMENT:
+        case REMAINDER:
+        case REMAINDER_ASSIGNMENT:
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: MAJOR Reduce this switch case number of lines from 6 to at most 5, for example by extracting code into methods. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1151)

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java [hunk 15]
```diff
@@ -0,0 +1,299 @@
...
+        default:
+          // do nothing
+      }
+    }
+
+    private void addUnderterminedConstraint(SymbolicValue sv, Tree tree) {
+      programState = programState.addConstraint(sv, new ZeroConstraint(tree, Status.UNDETERMINED));
+    }
+
+    private void checkOperation(Tree tree, SymbolicValue leftOp, SymbolicValue rightOp) {
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: MAJOR The Cyclomatic Complexity of this method "checkOperation" is 12 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java [hunk 16]
```diff
@@ -0,0 +1,299 @@
...
+import java.util.List;
+
+@Rule(key = "S3518")
+public class DivisionByZeroCheck extends SECheck {
+
+  private enum Status {
+    ZERO, NON_ZERO, UNDETERMINED;
+  }
+
+  private static class ZeroSymbolicValue extends SymbolicValue {
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: MAJOR Override this superclass' "equals" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2160)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* MAJOR 4 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [BinarySymbolicValue.java#L54](https://github.com/SonarSource/sonar-java/blob/3880cba88db93b02a0fbd6b8135426b6bb3e6463/java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/BinarySymbolicValue.java#L54): The Cyclomatic Complexity of this method "copyConstraint" is 11 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java [hunk 17]
```diff
@@ -0,0 +1,300 @@
...
+        default:
+          // do nothing
+      }
+    }
+
+    private void setAsUndetermined(SymbolicValue sv, Tree tree) {
+      programState = programState.addConstraint(sv, new ZeroConstraint(tree, Status.UNDETERMINED));
+    }
+
+    private void checkExpression(Tree tree, SymbolicValue leftOp, SymbolicValue rightOp) {
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: MAJOR The Cyclomatic Complexity of this method "checkExpression" is 12 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* MAJOR 4 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [BinarySymbolicValue.java#L54](https://github.com/SonarSource/sonar-java/blob/5e1d6d76682ff35c91d48e34c2409c09fbfe82a1/java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/BinarySymbolicValue.java#L54): The Cyclomatic Complexity of this method "copyConstraint" is 11 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java [hunk 18]
```diff
@@ -0,0 +1,300 @@
...
+import org.sonar.plugins.java.api.tree.TypeCastTree;
+import org.sonar.plugins.java.api.tree.UnaryExpressionTree;
+
+import java.util.List;
+
+@Rule(key = "S3518")
+public class DivisionByZeroCheck extends SECheck {
+
+  private enum Status {
+    ZERO, NON_ZERO, UNDETERMINED;
```
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: Two questions about this : what does it mean in terms of `zeroeness` if a SV has no constraint (thus no status) ?
How are things handled on Null/NotNull domain ?

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java [hunk 19]
```diff
@@ -0,0 +1,300 @@
...
+  }
+
+  private static class ZeroSymbolicValue extends SymbolicValue {
+
+    private final Status initialStatus;
+
+    public ZeroSymbolicValue(int id, Status initialConstraint) {
```
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: naming... constraint/status...

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java (hunk 18 above)
Comment (m-g-sonar) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: If the SV has no constraint, then we can't say anything as we didn't learn anything _yet_.
When the SV has the `UNDERTERMINED` constraint, we won't learn anything anymore.

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* MAJOR 4 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [BinarySymbolicValue.java#L54](https://github.com/SonarSource/sonar-java/blob/8444aa1a59c184762536d4f01db38dc944849463/java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/BinarySymbolicValue.java#L54): The Cyclomatic Complexity of this method "copyConstraint" is 11 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* MAJOR 4 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [BinarySymbolicValue.java#L54](https://github.com/SonarSource/sonar-java/blob/9b6885963334fc9da13c107788c789bfe1eb717d/java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/BinarySymbolicValue.java#L54): The Cyclomatic Complexity of this method "copyConstraint" is 11 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/BinarySymbolicValue.java [hunk 20]
```diff
@@ -58,7 +58,7 @@ public void computedFrom(List<SymbolicValue> symbolicValues) {
       return to.setConstraint(programState, shouldNotInverse().equals(booleanConstraint) ? boolConstraint : boolConstraint.inverse());
     } else if (constraintLeft instanceof ObjectConstraint) {
       ObjectConstraint nullConstraint = (ObjectConstraint) constraintLeft;
-      if (nullConstraint.isNull()) {
+      if (nullConstraint.isNull() || nullConstraint.isInversible()) {
```
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/BinarySymbolicValue.java]: Given implementation, the first operand is irrelevant here.

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java [hunk 21]
```diff
@@ -0,0 +1,300 @@
...
+        deferConstraint(Status.ZERO);
+      } else if (isNonZero(left) && isNonZero(right)) {
+        deferConstraint(Status.NON_ZERO);
+      }
+    }
+
+    private void handlePlusMinus(SymbolicValue left, SymbolicValue right) {
+      if (isZero(left) && isZero(right)) {
+        deferConstraint(Status.ZERO);
+      } else if ((isZero(left) && isNonZero(right)) || (isNonZero(left) && isZero(right))) {
```
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: This is in essence an assignement. As such this method should be simplified to handle the fact that if left is zero, then resulting SV is the right SV.

File: java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java [hunk 22]
```diff
@@ -0,0 +1,300 @@
...
+          deferConstraint(Status.NON_ZERO);
+        }
+      }
+    }
+
+    @Override
+    public void visitUnaryExpression(UnaryExpressionTree tree) {
+      if (!tree.is(Tree.Kind.LOGICAL_COMPLEMENT)) {
+        if (isZero(programState.peekValue())) {
+          deferConstraint(tree.is(Tree.Kind.UNARY_MINUS, Tree.Kind.UNARY_PLUS) ? Status.ZERO : Status.NON_ZERO);
```
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java]: could probably be simplified

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Pull request merged**